katdal/datasources.py

```diff
@@ -102,6 +103,80 @@ def _apply_data_lost(orig_flags, lost, block_id):
     return flags
 
 
+def corrprod_to_autocorr(corrprod):
```

Comment:
I'm really sensing the imminent birth of `utils.py` :-) This stuff would be very useful to `applycal` too.

Reply:
Makes sense. Should I keep it where it is for now and let you shuffle it around after this PR is merged?

Reply:
Yes, I think so. I still want to ponder the connections with `applycal` as well.